src/core/prometheus.cc

```diff
@@ -226,19 +227,39 @@ static std::string to_str(const seastar::metrics::impl::metric_value& v) {
     return ""; // we should never get here but it makes the compiler happy
 }
 
+/*
+ * Sanitizes the prometheus label value as per the line format rules and writes it out to the ostream:
+ * > label_value can be any sequence of UTF-8 characters, but the backslash (\), double-quote ("), and
+ * > line feed (\n) characters have to be escaped as \\, \", and \n, respectively.
+ */
+static void escape_and_write_label_value(std::ostream& s, std::string_view label_value) {
+    for (char c : label_value) {
+        switch (c) {
+            case '\\': s << "\\\\"; break;
+            case '\"': s << "\\\""; break;
+            case '\n': s << "\\n"; break;
+            default:   s << c;
+        }
```

Comment:
nitpick: since we're escaping escape characters

```suggestion
        switch (c) {
            case '\\': s << R"(\\)"; break;
            case '\"': s << R"(\")"; break;
            case '\n': s << R"(\n)"; break;
            default:   s << c;
        }
```

Reply:
> nitpick: since we're escaping escape characters

You could also use raw strings on the left, too.

Reply:
I'll adopt this when I upstream the change 👍